chore: enable 'return-await`

jrandolf-google opened the pull request:
With 'using', it's critical that we 'return await' instead of just returning. This ensures callbacks are run at the correct time. It's also more performant to 'return await' which is a plus.

OrKoN reviewed (approved):
LGTM % wait for tests to run

jrandolf committed:
chore: enable 'return-await`

With 'using', it's critical that we 'return await' instead of just returning. This ensures callbacks are run at the correct time. It's also more performant to 'return await' which is a plus.
